### PR TITLE
Adding SaveConfig().

### PR DESCRIPTION
**Pull request by g3rk6**
It would be nice to have a way to store configuration into a file. There could be better way of doing it, but this is the version that works for me.

### CODE REVIEW

File: viper.go [hunk 1]
```diff
@@ -874,6 +878,51 @@ func (v *Viper) InConfig(key string) bool {
...
+func SaveConfig() error { return v.SaveConfig() }
+func (v *Viper) SaveConfig() error {
+
+	jww.INFO.Println("Attempting to write config into the file.")
+	if !stringInSlice(v.getConfigType(), SupportedExts) {
+		return UnsupportedConfigError(v.getConfigType())
+	}
+
+	f, err := os.Create(v.getConfigFile())
+	defer f.Close()
```
Comment (s7v7nislands) [inline viper.go]: defer should after check error?

Comment (g3rk6) [inline viper.go]: Fixed it. Thanks for pointing it out!

**Comment by doublerebel** [comment]
LGTM :+1: .  Just needs docs.

**Comment by spf13** [comment]
Thanks for the contribution!
I love this idea, and the approach, but I think the PR needs some tweaking.

A few suggestions:

* Remove the debug line you left in
* Create a function in util.go that is called marshallConfigWriter
It would do the opposite of the unmarshallConfigReader function presently there
* Expand your current code to handle the additional types (hcl & prop)
* rename SaveConfig to WriteConfig()
* perhaps add another function called SafeWriteConfig() which only writes when the file doesn't currently exist.
* Add some tests. [Afero](https://github.com/spf13/afero) would be very helpful here. 
* Add some docs. It's a great feature, people should know about it.

I think you've done most of the work already, it's just mostly a question of reorganizing things.

Let me know if you want help or have questions. Happy to help.

**Comment by bryanl** [comment]
This idea is six months old, so I'd assume this PR is dead?

**Comment by AntonioMeireles** [comment]
@g3rk6 hi, anything (other than time) blocking you from progressing in this front ?

**Comment by g3rk6** [comment]
Sorry for the long delay. At the moment the only issue is time and currently super busy. I'll try to make some progress as soon as I get some time. Thanks.

**Comment by theherk** [comment]
@g3rk6 I'd love to see this fully implemented. I think it is looking good.

**Comment by spf13** [comment]
this is stale and it appears #287 has carried it forward. Closing this.

**Pull request closed without merging**